Services/DataBaseService/MigrationEngine.cs

```diff
@@ -0,0 +1,225 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data.SqlClient;
+using System.IO;
+using System.Text;
+using Microsoft.Extensions.Configuration;
+
+namespace DataBaseService
+{
+    public class MigrationEngine
+    {
+        private readonly IConfiguration configuration;
+
+        public MigrationEngine(IConfiguration configuration)
+        {
+            this.configuration = configuration;
+        }
+
+        /// <summary>
+        /// Migration method. 
+        /// Created "TradingStation" DB with "ExecutedScripts" table, then:
+        /// Runs through all the .sql in the directory (configure in the appsettings.json).
+        /// Checks (by its name) if a script was already executed, executes it, 
+        /// and marks as executed (by adding a row to the table).
+        /// </summary>
+        public void Migrate()
+        {
+            var connectionString = configuration.GetConnectionString("TradingStationString");
+            var scriptsToWriteDown = new Dictionary<string, ExecutedScript>
+            {
+                { "CreateDatabase", CreateDatabase() },
+                { "CreateScriptsTable", CreateScriptsTable(connectionString) }
+            };
+            List<string> scriptsToExecute = GetScriptsToExecute(connectionString);
+
+            using (var conn = new SqlConnection(connectionString))
+            {
+                conn.Open();
+                SqlTransaction executingTransaction = conn.BeginTransaction();
+                foreach (var fileName in scriptsToExecute)
+                {
+                    var scriptCode = File.ReadAllText(fileName);
+                    try
```

Comment:
move try catch before using, because we need all our scripts executed successfully and you can have exception on DB connection open

Reply:
Done. Now all the exceptions are catched. And we're still able to find out which script has failed: outer string used